Gems/Pointcloud/Code/Source/Render/PointcloudFeatureProcessor.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <Atom/RPI.Public/PipelineState.h>
+#include <Atom/RPI.Public/ViewportContextBus.h>
+#include <AzCore/Math/Transform.h>
+#include <Pointcloud/PointcloudFeatureProcessorInterface.h>
+namespace Pointcloud
+{
+    class Scene;
+    class Shader;
+
+    class PointcloudFeatureProcessor
+        : public PointcloudFeatureProcessorInterface
+        , protected AZ::RPI::ViewportContextIdNotificationBus::Handler
+        , protected AZ::Data::AssetBus::Handler
+    {
+    public:
+        AZ_RTTI(PointcloudFeatureProcessor, "{B6EF8776-F7F9-432B-8BD9-D43869FFFC3D}", PointcloudFeatureProcessorInterface);
+        AZ_CLASS_ALLOCATOR(PointcloudFeatureProcessor, AZ::SystemAllocator)
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        PointcloudFeatureProcessor() = default;
+        virtual ~PointcloudFeatureProcessor() = default;
+
+        // PointcloudFeatureProcessorInterface overrides
+        void SetTransform(const PointcloudHandle& handle, const AZ::Transform& transform) override;
+        void SetPointSize(const PointcloudHandle& handle, float pointSize) override;
+        PointcloudHandle AcquirePointcloud(const AZStd::vector<PointcloudAsset::CloudVertex>& cloudVertexData) override;
+        void SetVisibility(const PointcloudHandle& handle, bool visible) override;
+        void ReleasePointcloud(const PointcloudHandle& handle) override;
+
+    protected:
+        // RPI::SceneNotificationBus overrides
+        void OnRenderPipelineChanged(
+            AZ::RPI::RenderPipeline* pipeline, AZ::RPI::SceneNotification::RenderPipelineChangeType changeType) override;
+
+        // Data::AssetBus overrides
+        void OnAssetReloaded(AZ::Data::Asset<AZ::Data::AssetData> asset) override;
```

Comment:
It isn't enough if we haven't processed the asset yet. I think we should also implement OnAssetReady()